python_modules/dagster/dagster/_core/definitions/load_assets_from_modules.py

```diff
@@ -431,10 +447,32 @@ def assets_defs(self) -> Sequence[AssetsDefinition]:
             if isinstance(asset, AssetsDefinition) and asset.keys
         ]
 
+    @cached_property
+    def checks_defs(self) -> Sequence[AssetChecksDefinition]:
+        return [
+            cast(AssetChecksDefinition, asset)
+            for asset in self.loaded_objects
+            if isinstance(asset, AssetsDefinition) and has_only_asset_checks(asset)
+        ]
+
+    @cached_property
+    def assets_and_checks_defs(self) -> Sequence[Union[AssetsDefinition, AssetChecksDefinition]]:
+        return [*self.assets_defs, *self.checks_defs]
+
     @cached_property
     def source_assets(self) -> Sequence[SourceAsset]:
         return [asset for asset in self.loaded_objects if isinstance(asset, SourceAsset)]
 
+    @cached_property
+    def cacheable_assets(self) -> Sequence[CacheableAssetsDefinition]:
+        return [
+            asset for asset in self.loaded_objects if isinstance(asset, CacheableAssetsDefinition)
+        ]
+
+    @cached_property
+    def assets_only(self) -> Sequence[LoadableAssetTypes]:
```

Comment:
nit: maybe just call this `assets()`? nbd though

Reply:
I think it's confusing because AssetChecksDefinition is an assets_def, but yea naming isn't great. Planning on iterating on this one in downstreams potentially